### PR TITLE
CVE-2010-4476

### PR DESCRIPTION
**Pull request by kwc2894**
This was The apache tomcat issue which is delivered for part 1 with an extension till thursday night.

### CODE REVIEW

File: cves/CVE-2010-4476.yml [hunk 1]
```diff
@@ -43,24 +43,15 @@ description_instructions: |
...
-  A method used in the Java Runtime Environment, the one responsible for parsing
-  and converting double binary floating-point numbers, may lead to an infinite
-  loop of estimations while converting. An example of this is attempting to convert
-  the number 2.2250738585072012e-308 with the method.
-
-  This exploit can enable a denial of service from a remote attacker using a
-  crafted string that will trigger this infinite loop by causing a hang or a
-  repeatable crash on the affected programs.
-
+description: Attackers causing denial of cervis via a certain string to loop infinitely through estimations during a conversion of data to a floating number in some of the earlier versions of Java.
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: *service typo. I cant find where it says that this is a string input, it looks like a number. Could you double check that a string causes the loop?

File: cves/CVE-2010-4476.yml [hunk 2]
```diff
@@ -4,13 +4,13 @@ nickname_instructions: |
...
   alongside its CVE to make it more easily distinguished from the rest.
-nickname: Double Precision Floating-Point
+nickname: The GLUEY input
 
 CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE: 682
+CWE: 241
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: If this is caused by a number input, then I think the original CWE might actually work better.

File: cves/CVE-2010-4476.yml [hunk 3]
```diff
@@ -69,17 +60,12 @@ fixes_vcc_instructions: |
...
-      loop of trying to perfectly parse the decimal number.
+     note: This code contains the fix of the vulnerability and converts the string thats causing the forloop to be more readable data.
    - commit:
      note:
 vccs:
-  - commit: 407841c426dc52a4c6b8ccd297df6c484a540056
-    note: | 
-      SVN rev 302975, from the Tomcat website. This was the first commit
-      of this file in the directory; the file was added with the bug.
+  - commit:
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: I think there should be at least one VCC here

File: cves/CVE-2010-4476.yml [hunk 4]
```diff
@@ -99,7 +85,7 @@ upvotes_instructions: |
   upvotes to each vulnerability you see. Your peers will tell you how
   interesting they think this vulnerability is, and you'll add that to the
   upvotes score on your branch.
-upvotes: 4
+upvotes:
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: 2

File: cves/CVE-2010-4476.yml [hunk 5]
```diff
@@ -113,11 +99,9 @@ unit_tested:
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
     Must be just "true" or "false".
-  answer: |
-    There is no evidence in the Tomcat code that unit tests for these
-    files existed.
-  code: false
-  fix: false
+  answer: true
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: The answer is supposed to explain where you found or did not find the unit tests

File: cves/CVE-2010-4476.yml [hunk 6]
```diff
@@ -132,16 +116,9 @@ discovered:
...
-    While somewhat obscure, this was found by Rick Regan,a blog poster,
-    in that the number 2.2250738585072011E-308 would hang a PHP application
-    while it tried to process the number, around the week of January 6, 2011.
-    On February 3rd, 2011, Bryan Sullivan from the Adobe Secure Software
-    Development Team discovered that this same number would cause a hang in Java
-    applications as well. There is a very small range of numbers that can cause
-    this hang due to the parsing of the number.
-  date: 2011-02-03
-  automated: false
+  answer:
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: The answer is supposed to explain where you found the date

File: cves/CVE-2010-4476.yml [hunk 7]
```diff
@@ -151,8 +128,8 @@ subsystem:
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
     Examples: "clipboard", "gpu", "ssl", "speech", "renderer"
-  answer: Based on the path affected by the fix commit.
-  name: catalina
+  answer:  container/catalina/src/share/org/apache/catalina/connector/Request.java
+  name: src, share, org, apache, catalina, connector, request
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: Which one of these is the actual subsystem?

File: cves/CVE-2010-4476.yml [hunk 8]
```diff
@@ -162,13 +139,10 @@ interesting_commits:
...
     section by explaining what happened between the VCCs and the fix.
-  answer: |
-    This bug went unnoticed for 7 years, so a majority of the commits in
-    the subsystem occured from the bug's inception, in 2004 to the fix in 2011.
-    The bug was introduced in the first commit for the directory.
+  answer:
   commits:
-    - commit:
-      note:
+    - commit:  container/catalina/src/share/org/apache/catalina/connector/Request.java
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: This is supposed to contain the commit hash

File: cves/CVE-2010-4476.yml [hunk 9]
```diff
@@ -162,13 +139,10 @@ interesting_commits:
     emerging themes?
     If there are no interesting commits, demonstrate that you completed this
     section by explaining what happened between the VCCs and the fix.
-  answer: |
-    This bug went unnoticed for 7 years, so a majority of the commits in
-    the subsystem occured from the bug's inception, in 2004 to the fix in 2011.
-    The bug was introduced in the first commit for the directory.
+  answer:
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: If there are no interesting commits then you should explain why that is true under this section

File: cves/CVE-2010-4476.yml [hunk 10]
```diff
@@ -185,40 +159,37 @@ lessons:
...
-    applies:
+    applies: false 
     note:
   distrust_input:
     applies: true
-    note: |
-      It's difficult to just say to not trust the input parsed, as the bug is
-      an edge case, but implementing a failsafe for potential inputs, as done in
-      the fix, would have solved the potential of an issue from the beginning.
+    note: This is distrusted input because it accounts for a particular type of int that exploits a double loop
```
Comment (dajanio) [inline cves/CVE-2010-4476.yml]: I think that it converts floating point numbers, could you double check this.

File: cves/CVE-2010-4476.yml (hunk 3 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: when i origionally opened the file it didnt have one but from the split view im viewing this in i am now seeing one. I will add it in.

File: cves/CVE-2010-4476.yml (hunk 5 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: Im sorry, then why do the instructions say must be true or false?

File: cves/CVE-2010-4476.yml (hunk 6 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: Ah. Okay. i will go back and see where i got the date then

File: cves/CVE-2010-4476.yml (hunk 7 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: i believe they are all subsystems affected and this allowed me to list them out

File: cves/CVE-2010-4476.yml (hunk 10 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: I will and provide a follow up comment

**Review by kwc2894 — COMMENTED** [review]
This is my initial response, i will be investigating shortly

File: cves/CVE-2010-4476.yml (hunk 9 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: I will take that to note

File: cves/CVE-2010-4476.yml (hunk 2 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: Made the fix. thank you for that.

File: cves/CVE-2010-4476.yml (hunk 1 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: So turns out the CVE said it was a string although they do input a number into it. I am gonna just say its a number input.

File: cves/CVE-2010-4476.yml (hunk 8 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: ah that was a mistake.

File: cves/CVE-2010-4476.yml (hunk 10 above)
Comment (kwc2894) [reply cves/CVE-2010-4476.yml]: it converts floating point numbers

**Review by kwc2894 — COMMENTED** [review]
reviewed input

**Pull request merged**